includes/activity/class-generic-object.php

```diff
@@ -297,8 +297,11 @@ public function to_array( $include_json_ld_context = true ) {
 				$value = $value->to_array( false );
 			}
 
-			// If value is still empty, ignore it for the array and continue.
-			if ( isset( $value ) ) {
+			if ( $this->is_namespaced( $key ) && is_array( $value ) ) {
```

Comment:
I wonder if swapping the two checks could be a micro-optimization? We wouldn't have to check keys where the value is not an array.